### ISSUE TITLE
VS2022 support?

### ISSUE THREAD

**Issue by anupshah** [mmanela/chutzpah#807]
Hi, thanks for this extension.

Any plans for supporting VS2022?

**Comment by acangialosi** [comment]
@mmanela, please let us know if you have any issues upgrading the extension to VS 2022. We can loop in folks from the engineering team to help out with any questions or technical issues. Do you have an idea of when you expect to publish your new version?

**Comment by DomZZ** [comment]
@mmanela, please make your adapter working on VS2022.
Chutzpah apdater/runner are the only things that prevent us to move on VS 2022 :(

**Comment by mmanela** [comment]
Thanks for reaching out @DomZZ  @acangialosi . I would love to release for VS 2022 but just have not had the time. The process looks more involved than for past releases so I will need to find time once things settle down at work/home. If I get a PR the changes, it would make the process fast but barring that it make take a bit of time.

**Comment by iprak** [comment]
@mmanela The structure and dependencies of the extension make its side by side migration to VS2022 difficult. I took the approach of creating just VS2022 specific extension and have had good success with the extension itself. I haven't been able to get unit tests working though; I am not sure if this is due change in Xunit or something else. Unfortunately this not something which can be merged into main repo so I did not submit a PR but here is the code https://github.com/iprak/chutzpah/tree/Vs2022

**Comment by mmanela** [comment]
@iprak  Thanks for taking a start on this. Can you share details on the changes you needed to make? I also noticed you commented out some code and was curious why?

**Comment by iprak** [comment]
I think I only commented out cs code in 2 files (vstst.cs and ChutzpahTestSettingsFile.cs); the code did not seem used and was flagged in CodeAnalysis. Yes I did comment out Fact.Testable > Inject; I just could not get it compiling :-(

Here is a summary of what I settled out after some iterations:

High level changes:
- Upgraded projects to .net 48 and removed unused settings, etc.
- Got rid of all individual package references (e.g. packages.config) and switched to PackageReference 
- Removed fingerprint,etc. from nuget.config .. they were preventing some new packages from being installed.
- Adjusted paths for utilities/components as necessary.

Previously, Chutzpah.csproj listed some packages e.g. psake, xunit.console.runner so that they can be used in psake tasks. Some references were strong signed via a psake task and deposited in packages_autosigned folder. These signed components were explcitly referenced from packages_autosigned folder.

We can't do this with PackageReference. I decided to let packages be installed at their default location instead of forcing them into local packages folder. This makes best use of user nuget cache.

Build.bat was instead adjusted to install these utility packages, psake being one of them.

At the end I renamed all VS2012 instances to VS2022 since that was just confusing to me.
